### ISSUE TITLE
WalletsIn.java:70-73: Return the new instance of the...

### ISSUE THREAD

**Issue by 0pdd** [zold-io/java-api#12]
The puzzle `4-968ad704` from #4 has to be resolved:

https://github.com/zold-io/java-api/blob/93fa5fc251ff69dcb28db1b9a5195c111b05c004/src/main/java/io/zold/api/WalletsIn.java#L70-L73

The puzzle was created by Tolegen Izbassar on 03-Jul-18. 

Estimate: 30 minutes,     

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0pdd** [comment]
I can't create GitHub labels `pdd`. Most likely I don't have necessary permissions to `zold-io/java-api` repository. Please, make sure @0pdd user is in the [list of collaborators](https://github.com/zold-io/java-api/settings/collaboration):

```Octokit::NotFound
POST https://api.github.com/repos/zold-io/java-api/labels: 404 - Not Found // See: https://developer.github.com/v3/issues/labels/#create-a-label
/app/vendor/bundle/ruby/2.3.0/gems/octokit-4.9.0/lib/octokit/response/raise_error.rb:16:in `on_complete'
/app/vendor/bundle/ruby/2.3.0/gems/faraday-0.15.2/lib/faraday/response.rb:9:in `block in call'
/app/vendor/bundle/ruby/2.3.0/gems/faraday-0.15.2/lib/faraday/response.rb:61:in `on_complete'
/app/vendor/bundle/ruby/2.3.0/gems/faraday-0.15.2/lib/faraday/response.rb:8:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/octokit-4.9.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
/app/vendor/bundle/ruby/2.3.0/gems/octokit-4.9.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/faraday-0.15.2/lib/faraday/rack_builder.rb:143:in `build_response'
/app/vendor/bundle/ruby/2.3.0/gems/faraday-0.15.2/lib/faraday/connection.rb:387:in `run_request'
/app/vendor/bundle/ruby/2.3.0/gems/faraday-0.15.2/lib/faraday/connection.rb:175:in `post'
/app/vendor/bundle/ruby/2.3.0/gems/sawyer-0.8.1/lib/sawyer/agent.rb:94:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/octokit-4.9.0/lib/octokit/connection.rb:156:in `request'
/app/vendor/bundle/ruby/2.3.0/gems/octokit-4.9.0/lib/octokit/connection.rb:28:in `post'
/app/vendor/bundle/ruby/2.3.0/gems/octokit-4.9.0/lib/octokit/client/labels.rb:44:in `add_label'
/app/objects/github_tagged_tickets.rb:50:in `block in submit'
/app/objects/github_tagged_tickets.rb:50:in `each'
/app/objects/github_tagged_tickets.rb:50:in `submit'
/app/objects/commit_tickets.rb:38:in `submit'
/app/objects/emailed_tickets.rb:35:in `submit'
/app/objects/sentry_tickets.rb:46:in `submit'
/app/objects/puzzles.rb:91:in `block in expose'
/app/objects/puzzles.rb:82:in `loop'
/app/objects/puzzles.rb:82:in `expose'
/app/objects/puzzles.rb:33:in `deploy'
/app/objects/job.rb:38:in `proceed'
/app/objects/job_starred.rb:33:in `proceed'
/app/objects/job_recorded.rb:32:in `proceed'
/app/objects/job_emailed.rb:35:in `proceed'
/app/objects/job_commiterrors.rb:36:in `proceed'
/app/objects/job_detached.rb:48:in `exclusive'
/app/objects/job_detached.rb:36:in `block in proceed'
/app/objects/job_detached.rb:36:in `fork'
/app/objects/job_detached.rb:36:in `proceed'
/app/0pdd.rb:342:in `block in <top (required)>'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1634:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1634:in `block in compile!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:992:in `block (3 levels) in route!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1011:in `route_eval'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:992:in `block (2 levels) in route!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1039:in `block in process_route'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1037:in `catch'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1037:in `process_route'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:990:in `block in route!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:989:in `each'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:989:in `route!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1096:in `block in dispatch!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1075:in `block in invoke'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1075:in `catch'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1075:in `invoke'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1093:in `dispatch!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:924:in `block in call!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1075:in `block in invoke'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1075:in `catch'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1075:in `invoke'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:924:in `call!'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:913:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-2.0.1/lib/rack/protection/xss_header.rb:18:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-2.0.1/lib/rack/protection/path_traversal.rb:16:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-2.0.1/lib/rack/protection/json_csrf.rb:26:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-2.0.1/lib/rack/protection/base.rb:50:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-2.0.1/lib/rack/protection/base.rb:50:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-2.0.1/lib/rack/protection/frame_options.rb:31:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-2.0.5/lib/rack/logger.rb:15:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-2.0.5/lib/rack/common_logger.rb:33:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:231:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:224:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-2.0.5/lib/rack/head.rb:12:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-2.0.5/lib/rack/method_override.rb:22:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:194:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1957:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1501:in `block in call'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1728:in `synchronize'
/app/vendor/bundle/ruby/2.3.0/gems/sinatra-2.0.1/lib/sinatra/base.rb:1501:in `call'
/app/vendor/bundle/ruby/2.3.0/gems/rack-2.0.5/lib/rack/handler/webrick.rb:86:in `service'
/app/vendor/ruby-2.3.3/lib/ruby/2.3.0/webrick/httpserver.rb:140:in `service'
/app/vendor/ruby-2.3.3/lib/ruby/2.3.0/webrick/httpserver.rb:96:in `run'
/app/vendor/ruby-2.3.3/lib/ruby/2.3.0/webrick/server.rb:296:in `block in start_thread'
```

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@6ac9e0a3-eadb-45bf-b051-800033971ed4, thread: pool-609-thread-1
com.zerocracy.radars.github.ThrottledComments[76] java.lang.IllegalStateException: Can't post an identical message to zold-io/java-api#12: @llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue
```

[0.24.9](https://github.com/zerocracy/farm): Issue: zold-io/java-api#12, Action: `opened`

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/zold-io/java-api/issues/12#issuecomment-402844617))

@llorllale Job [#12](https://github.com/zold-io/java-api/issues/12) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#12](https://github.com/zold-io/java-api/issues/12) assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/CAZUREFND/1862062215); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by llorllale** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zold-io/java-api/issues/12#issuecomment-402882356))

@llorllale The user @llorllale[/z](https://www.0crat.com/u/llorllale) resigned from [#12](https://github.com/zold-io/java-api/issues/12), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
The job [#12](https://github.com/zold-io/java-api/issues/12) assigned to @t-izbassar[/z](https://www.0crat.com/u/t-izbassar), here is [why](https://www.0crat.com/footprint/CAZUREFND/1911815061); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@t-izbassar[/z](https://www.0crat.com/u/t-izbassar) this job was assigned to you 6days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @t-izbassar[/z](https://www.0crat.com/u/t-izbassar) resigned from [#12](https://github.com/zold-io/java-api/issues/12), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @t-izbassar[/z](https://www.0crat.com/u/t-izbassar)

**Comment by 0crat** [comment]
The job [#12](https://github.com/zold-io/java-api/issues/12) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CAZUREFND/2021947028); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by victornoel** [comment]
@llorllale something is not clear, how do you give an id to the created wallet?
Either `create` should take an id (`long`) or it should generate a random one.
Generating a random one could cause problems in case of conflict with an existing wallet I suppose.

**Comment by victornoel** [comment]
@0crat waiting for an answer to https://github.com/zold-io/java-api/issues/12#issuecomment-406849015

**Comment by 0crat** [comment]
> @0crat waiting for an answer to https://github.com/zold-io/java-api/issues/12#issuecomment-406849015 ([here](https://github.com/zold-io/java-api/issues/12#issuecomment-406849029))

@victornoel The impediment for [#12](https://github.com/zold-io/java-api/issues/12) was registered successfully by @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by victornoel** [comment]
@llorllale ping :)

**Comment by victornoel** [comment]
@llorllale I will go on with an id passed as an argument, and we can iterate on that in the PR.

**Comment by victornoel** [comment]
@0pdd this is implemented in #52

**Comment by 0pdd** [comment]
> @0pdd this is implemented in #52

I see you're talking to me, but I can't reply since I'm not a chat bot.

**Comment by llorllale** [comment]
@victornoel this is how it's done in the ruby API: 

https://github.com/zold-io/zold/blob/81af0dc81d81e060282a89573881da7b752b4d89/lib/zold/id.rb#L29-L37

**Comment by 0pdd** [comment]
The puzzle `4-968ad704` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#65](https://github.com/zold-io/java-api/issues/65), [#66](https://github.com/zold-io/java-api/issues/66) are still not solved.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#12](https://github.com/zold-io/java-api/issues/12) is now out of scope

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0pdd** [comment]
@0pdd the puzzle [#65](https://github.com/zold-io/java-api/issues/65) is still not solved; solved: [#66](https://github.com/zold-io/java-api/issues/66).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#85](https://github.com/zold-io/java-api/issues/85), [#86](https://github.com/zold-io/java-api/issues/86) are still not solved; solved: [#65](https://github.com/zold-io/java-api/issues/65), [#66](https://github.com/zold-io/java-api/issues/66).
